exporters/prometheus/exporter_test.go

```diff
@@ -1006,19 +1014,26 @@ func TestDuplicateMetrics(t *testing.T) {
 
 			tc.recordMetrics(ctx, meterA, meterB)
 
-			match := false
-			for _, filename := range tc.possibleExpectedFiles {
-				file, ferr := os.Open(filename)
-				require.NoError(t, ferr)
-				t.Cleanup(func() { require.NoError(t, file.Close()) })
-
-				err = testutil.GatherAndCompare(registry, file)
-				if err == nil {
-					match = true
-					break
+			if tc.expectGatherError {
+				// With improved error handling, conflicting instrument types emit an invalid metric.
+				// Gathering should surface an error instead of silently dropping.
+				_, err := registry.Gather()
```

Comment:
Do you know if there is a way for us to still compare the output, despite the error?  Some users may switch to continue on errors, so it would be good to still test that we get the desired output if possible.